File: 944. Delete Columns to Make Sorted.cpp
class Solution {
public:
    int minDeletionSize(vector<string>& str) 
    {
        map<int,string>mp;
        int i = 0, n = str[0].size();
        while(i<n)
        {
            for(auto s:str)
                mp[i] += (s[i]);
            i++;
        }

        int ans = 0;
        for(auto [val, s]:mp)
        {
            string temp = s;
            sort(temp.begin(), temp.end());
            if(temp != s) ans++;
        }
        return ans;
    }
};

File: String/944. Delete Columns to Make Sorted.py
class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        n, m, ans = len(strs), len(strs[0]), 0
        for j in range(m):
            store = []
            for i in range(n):
                store.append(strs[i][j])
            ans += store != sorted(store)
        return ans